R/adducts.R

```diff
@@ -1,18 +1,47 @@
-#' @title Create list with all adduct calculation rules (all rules)
-#' 
-#' This function returns a list with rules for the calculation of adducts. 
-#' It is required for the calculation of adduct m/z values from exact masses 
-#' and the other way round. This list contains all rules for both ionization 
-#' modes.
-#' 
-#' 
-#' @author Michael Witting, \email{michael.witting@@helmholtz-muenchen.de}
-#'
-#' @seealso \code{\link{adduct_rules_pos}}
-#' @seealso \code{\link{adduct_rules_neg}}
-#' @seealso \code{\link{get_adduct_names}}
-#'
-#' @export
+mass2mz <- function(exact_mass, adduct) {
+  
+  # get all rules for adduct calculation
+  adduct_rules_all <- adduct_rules_all()
+  
+  # check if adduct suppplied is in the list of valid adducts
+  if(!adduct %in% names(adduct_rules_all)) {
+    
+    stop(paste0("Unknown adduct: ", adduct))
+    
+  }
+  
+  # retrieve multiplicative and additive part for calculation
+  mass_multi <- adduct_rules_all[[adduct]]$mass_multi
+  mass_add <- adduct_rules_all[[adduct]]$mass_add
+  
+  ion_mass <- exact_mass * mass_multi + mass_add
+  
+  return(ion_mass)
```

Comment:
`return` should not be required. The default in `R` is to simply use here `ion_mass`.